Marlin/Marlin_main.cpp

```diff
@@ -5922,18 +5929,17 @@ inline void gcode_M17() {
   }
 
   static void ensure_safe_temperature() {
-    bool did_show = false;
     wait_for_heatup = true;
     while (wait_for_heatup) {
       idle();
+      if (!wait_for_heatup) break;
```

Comment:
I added a new commit to resolve this in a different way that will prevent any possible race conditions or logic errors associated with `wait_for_heatup` being reset by M108 in the emergency parser. This should allow M108 to work as desired with M600.